app/code/Magento/Customer/Block/Widget/City.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Copyright 2025 Adobe
+ * All Rights Reserved.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Customer\Block\Widget;
+
+use Magento\Customer\Api\AddressMetadataInterface;
+use Magento\Customer\Api\CustomerMetadataInterface;
+use Magento\Customer\Api\Data\AttributeMetadataInterface;
+use Magento\Customer\Api\Data\CustomerInterface;
+use Magento\Customer\Helper\Address as AddressHelper;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Element\Template\Context;
+
+/**
+ * Widget for showing customer city.
+ *
+ * @method CustomerInterface getObject()
+ * @method Name setObject(CustomerInterface $customer)
+ *
+ * @SuppressWarnings(PHPMD.DepthOfInheritance)
+ */
+class City extends AbstractWidget
+{
+
+    /**
+     * the attribute code
+     */
+    public const ATTRIBUTE_CODE = 'city';
+
+    /**
+     * @param Context $context
+     * @param AddressHelper $addressHelper
+     * @param CustomerMetadataInterface $customerMetadata
+     * @param AddressMetadataInterface $addressMetadata
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        AddressHelper $addressHelper,
+        CustomerMetadataInterface $customerMetadata,
+        private readonly AddressMetadataInterface $addressMetadata,
+        array $data = []
+    ) {
+        parent::__construct($context, $addressHelper, $customerMetadata, $data);
+        $this->_isScopePrivate = true;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function _construct(): void
+    {
+        parent::_construct();
+
+        // default city location
+        $this->setTemplate('Magento_Customer::widget/city.phtml');
+    }
+
+    /**
+     * Can show prefix
+     *
+     * @return bool
+     * @throws LocalizedException
+     */
+    public function showCity()
```

Comment:
Specify the return type 
```suggestion
    public function showCity():bool
```